repos/system_upgrade/common/libraries/rhsm.py

```diff
@@ -383,6 +384,44 @@ def switch_certificate(context, rhsm_info, cert_path):
             context.copy_to(cert_path, os.path.join(path, os.path.basename(cert_path)))
 
 
+def is_rhsm_registered(context):
+    """
+    Check whether the system is registered with Red Hat Subscription Manager
+
+    Note that this doesn't differentiate between SCA and SKU access
+
+    :param context: An instance of a mounting.IsolatedActions class
+    :type context: mounting.IsolatedActions class
+    :return: True if it is registered, false otherwise
+    :rtype: bool
+    """
+    try:
+        result = context.call(['subscription-manager', 'identity', '--no-progress-messages'], checked=False)
+    except OSError as e:
+        api.current_logger().error('Failed to execute subscription-manager executable')
+        raise StopActorExecutionError(
+            message='Unable to execute subscription-manager executable: {}'.format(str(e)),
+            details={
+                'hint': 'Please ensure subscription-manager is installed and executable.'
+            }
+        )
```

Comment:
@matejmatuska @ptoscano subscription manager does not have to be installed on the system. systems that are not subscribed do not have to have subscription-manager installed. so this should function should respect execution with `--no-rhsm` and in such a case should not raise an error